### PR TITLE
Check hardware for AP and if so record 'can_be_ap = True' to /etc/iiab/iiab.ini

### PR DESCRIPTION
**Pull request by holta**
Tested on 5 different hardware scenarios.

Refines the following PR, which also needed a couple Ansible fixes:

- PR #3070

### CODE REVIEW

**Comment by holta** [comment]
@jvonau is this PR hopefully good enough, in your opinion?

**Comment by jvonau** [comment]
#3216 has the whole test series, why was that one not updated, are you not considering the other 2 pending PRs?

**Comment by tim-moody** [comment]
Is anyone worried that a user will type iiab-hotspot on and get a file not found error rather than being told that their system does not support a hotspot?

**Comment by jvonau** [comment]
'Release Notes' should be altered to explain why that command may not be found and where to look in the iiab.ini file to confirm what was or not detected. I'm not concerned as that should prompt a question from the user for further clarification or the user could figure that out from the 'can_be_ap' entry in iiab.ini should there some familiarity with the code base or adequate notes available.

**Comment by jvonau** [comment]
> Tested on 5 different hardware scenarios.
> 

Could I be so bold as to ask which 5 and are their iiab-diagnostics available for review?
> Refines the following PR, which also needed a couple Ansible fixes:
> 

Were altered in the roll-up PR.

**Comment by holta** [comment]
> are you not considering the other 2 pending PRs?

I definitely want to work through these 2 PR's also:

- PR #3173
- PR #3179

**Comment by holta** [comment]
> Is anyone worried that a user will type iiab-hotspot on and get a file not found error rather than being told that their system does not support a hotspot?

It's a great question: in the sense that this certainly needs to documented prominently (somewhere!) so that the answer's right in your face (or worst case loudly advertise a URL, explaining better the details?)

1) Should [iiab-hotspot-on](https://github.com/iiab/iiab/blob/master/roles/network/templates/network/iiab-hotspot-on) explain better, in similar such circumstances?

2) Or is it rather too late at that point &mdash; maybe this coming-to-terms with machine/topology ground-truths should happen earlier on &mdash; e.g. is this more the job of things like ./iiab-network i.e. roles/network/tasks/main.yml ?

**Comment by jvonau** [comment]
Guess providing the roll-up PR was a waste of my time.

**Comment by holta** [comment]
> 'Release Notes' should be altered to explain why that command may not be found and where to look in the iiab.ini file to confirm what was or not detected.

Yes we should start there.

But "nobody reads" in this era (unless they're absolutely forced to!)

So we should "point their nose" precisely in the right direction &mdash; e.g. in Ansible's output and/or `iiab-hotspot-on` output &mdash; in plain English as much as possible.

**Comment by holta** [comment]
> Guess providing the roll-up PR was a waste of my time.

I'll look at it too.  I thought it was equivalent to #3070 + #3173 + #3179 combined into one PR.  But if I misunderstood, i.e. that it also includes subsequent improvements, I apologize.

**Comment by jvonau** [comment]
> But "nobody reads" in this era (unless they're absolutely forced to!)
> 
> So we should "point their nose" precisely in the right direction — e.g. in Ansible's output and/or `iiab-hotspot-on` output — in plain English as much as possible.

Then toss this entire PR and place the detection in iiab-hotspot-on

**Comment by jvonau** [comment]
> > Guess providing the roll-up PR was a waste of my time.
> 
> I'll look at it too. I thought it was equivalent to #3070 + #3173 + #3179 combined into one PR. But if I misunderstood, i.e. that it also includes subsequent improvements, I apologize.

I clearly referenced what changes for 3070 where done within the roll up.

**Comment by holta** [comment]
> > Tested on 5 different hardware scenarios.

> which 5

Age-old Intel NUC's mostly, each of which had their own weird output when running things like `iw list | grep AP`

Some more long-winded than others.

I can re-run several of those examples and paste the relevant output into this ticket.

**Comment by holta** [comment]
This new feature is summarized at:
https://twitter.com/internet_in_box/status/1527690188637540352

**Comment by holta** [comment]
Thanks to @jvonau whose PR #3179 further cleaned up here:

https://github.com/iiab/iiab/blob/35750850ac3c387b3a356371440ff48138f64123/roles/network/templates/hostapd/iiab-hotspot-on#L2-L6
